Add DAG for airflow metadb cleanup for Composer.

maciekgawron opened the pull request:
## Description

Added DAG for cleaning airflow metadb for Composer.
The DAG should be ready to use, just put the DAG in the Composer bucket.
The default retention period is 30days, it can be changed using airflow_db_cleanup__max_db_entry_age_in_days airflow variable.

The DAG is a fork from open-source repository https://github.com/teamclairvoyant/airflow-maintenance-dags/tree/master/db-cleanup


## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

leahecole reviewed (changes requested):
Hey Maciek! First of all, this is a great addition to our repo fo sure. A couple of questions to help make sure it is up to repo standards and also has the reach we want

- Could you add DAG testing like all of our other DAGs? [This is an example of a test for a DAG that set variables](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/composer/workflows/dataflowtemplateoperator_tutorial_test.py), like this one would need. The imports might look a little different which leads to my next bullet point...
- Please put this directory in `workflows`. If you felt it should be in `tools`, I'd be fine with that too - `tools` is actually empty - before we had in place image upgrades, there was a migration script, but that's been gone since before the WAW team worked on Composer :) 


Once you've added a test, take a look at the [test environment setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup) if you'd like to try running the tests locally. 

most importantly, if you need any help with running tests, please ping me, or set up some time early California morning and I can help guide this PR to the finish.

maciekgawron commented:
Hi Leah,

I moved the dag to workflows and added a test.
I managed to test it locally but it required locally running airflow+mysql.
I wonder if it is possible to test the DAGs in composer environment set in test-env.sh.

leahecole commented:
Hey! Sorry for delay. If you don't have a personal project you're testing against, the best bet is to use CI/CD. If you're not part of the GoogleCloudPlatform github org, the tests won't trigger. I just triggered them manually for you - if you ever need me to kick off a job, you can always ping me even if it's the middle of the night I'll kick it off first thing the next day :)

leahecole commented on composer/functions/composer_storage_trigger_test.py:
```diff
@@ -11,12 +11,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import composer_storage_trigger
```
this change needs to be reverted and lint will then pass

maciekgawron replied:
Interesting. With my local nox run I got:

```
./functions/composer_storage_trigger_test.py:18:1: I100 Import statements are in the wrong order. 'import composer_storage_trigger' should be before 'import pytest'
import composer_storage_trigger
```

That is why I moved the import. But if this fails ci/cl I am reverting the change.

leahecole reviewed (changes requested):
I noticed that you removed the README - is this going to be included or referenced in a tutorial? Or is the intent to share it with customers as needed? Just want to make sure this is discoverable and usable by folks who stumble upon it

maciekgawron commented:
> I noticed that you removed the README - is this going to be included or referenced in a tutorial? Or is the intent to share it with customers as needed? Just want to make sure this is discoverable and usable by folks who stumble upon it

I moved the content of README to the top of the DAG file. There is already a general README.rst in this catalogue, not sure if the other one will fit with the structure. Where do you usually put instructions for the other workflows?

leahecole commented:
My bad - did not notice it had gone into the DAG. Most of these DAGs are embedded in tutorials on cloud.google.com, but some are not. Up to you for how you think folks should be pointed to it

leahecole commented:
working on trying to get the CI to run. Not sure why 3.6 is the only pending one 🤦‍♀️
